src/components/layout/genericDialog/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
+
+interface IProps {
+    open: boolean,
+    title: string,
+    description: string,
+    fields: Array<any>,
+    handleChange: any,
+    handleSubmit: any,
+    handleClose: any
+}
+
+function GenericDialog({ open, title, description, fields = [], handleChange, handleSubmit, handleClose }: IProps): JSX.Element {
+    return (
+        <Dialog open={open} onClose={() => "handleClose"} aria-labelledby="form-dialog-title">
```

Comment:
I dont understand the `onClose` callback
```suggestion
        <Dialog open={open} aria-labelledby="form-dialog-title">
```

Reply:
My mistake,

should be onClose={handleClose} to enable the ESC key